packages/clarity-js/src/interaction/timeline.ts

```diff
@@ -0,0 +1,44 @@
+import { Event } from "@clarity-types/data";
+import { TimelineState } from "@clarity-types/interaction";
+import config from "@src/core/config";
+import * as envelope from "@src/data/envelope";
+import encode from "@src/interaction/encode";
+
+let state: TimelineState[] = [];
+export let updates: TimelineState[] = [];
+
+export function start(): void {
+    state = [];
+    reset();
+}
+
+export function reset(): void {
+    updates = [];
+}
+
+export function track(time: number, event: Event, target:number, x: number, y: number): void {
+    state.push({
+        time,
+        event: Event.Timeline,
+        data: { type: event, target, x, y }
+    });
+}
+
+export function compute(): void {
+    updates = [];
+    let max = envelope.data.start + envelope.data.duration;
+    let min = Math.max(max - config.timeline, 0);
```

Comment:
this seems like we are sending data for current payload and not previous payload

Reply:
Clarified offline.